src/mprk.jl

```diff
@@ -49,131 +49,72 @@ function build_mprk_matrix(P, sigma, dt)
 end
 
 # in-place for dense arrays
-function build_mprk_matrix!(M, a, P, D, sigma, dt)
+function build_mprk_matrix!(M, P, sigma, dt)
```

Comment:
This is now the same implementation as in the core of the out-of-place version above, isn't it? If so, can we rewrite the out-of-place version as

```julia
# out-of-place for dense and static arrays
function build_mprk_matrix(P, sigma, dt)
    # re-use the in-place version implemented below
    M = similar(P)
    build_mprk_matrix!(M, P, sigma, dt)
    
    if P isa StaticArray
        return SMatrix(M)
    else
        return M
    end
end
```

Reply:
I think you're right. I'll change this.